src/main/webapp/js/index.js

```diff
@@ -70,24 +88,38 @@ function createReceiptCardElement(receipt) {
 
   // Attach receipt card clone to parent div.
   document.getElementById('receipts-display').appendChild(receiptCardClone);
+
+  // Attach listener to trigger the deletion of this receipt.
+  attachDeleteButtonEventListener(receipt);
 }
 
 /**
- * Creates error message based on existing HTML template.
- * @param {string} label User-entered label.
+ * Attaches event listener to delete button.
+ * @param {Receipt} receipt A Receipt datastore object.
  */
-function createErrorMessageElement(label) {
-  // Clone error message from template.
-  const errorMessageClone =
-      document.querySelector('#error-message-template').content.cloneNode(true);
-
-  // Fill in template fields with correct information.
-  errorMessageClone.querySelector('h3').innerText =
-      'Sorry, no results found for "' + label +
-      '". Please try your search again or try a different query.';
+function attachDeleteButtonEventListener(receipt) {
+  // Get all delete buttons; most recently attached will be at the end.
+  const allDeleteButtons =
+      document.querySelector('#receipts-display').querySelectorAll('#delete');
+  const mostRecentDeleteButton = allDeleteButtons[allDeleteButtons.length - 1];
```

Comment:
Maybe including `receiptCardClone` as an argument and using a query selector on it would be more efficient than getting all the delete buttons.

Reply:
receiptCardClone is a DocumentFragment, so after it's attached to the DOM, the DocumentFragment is left empty. Trying to access the delete button from receiptCardClone would result in a null exception on the click event

https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment#usage_notes

Reply:
Could the receipt card element contain the receipt id? Then that can be used to locate the receipt to remove from the DOM.

Reply:
The `appendChild()` call above should return the node that was appended so you could pass that in into` attachDeleteButtonEventListener()` and then query the delete button within that node. Something like:
```
  // Attach receipt card clone to parent div.
  const node = document.getElementById('receipts-display').appendChild(receiptCardClone);

  // Attach listener to trigger the deletion of this receipt.
  attachDeleteButtonEventListener(node, receipt);
```

Reply:
@acheng1 I tried the appendChild() method but according to the documentation: 

> The returned value is the appended child (aChild), except when aChild is a DocumentFragment, in which case the empty DocumentFragment is returned.

So the issue of the node being null on the button click would still exist. I found Bryan's suggestion worked well though!